crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -602,11 +604,11 @@ pub enum BinaryOp {
     Mul, //(*)
     #[allow(dead_code)]
     SafeMul, //(*) safe multiplication
-    Udiv, //(/) unsigned division
-    Sdiv, //(/) signed division
-    Urem, //(%) modulo; remainder of unsigned division
-    Srem, //(%) remainder of signed division
-    Div, //(/) field division
+    Udiv(Option<Location>), //(/) unsigned division
+    Sdiv(Location), //(/) signed division
```

Comment:
Why does `Udiv` have an optional location while the other variants have mandatory locations? Even `Sdiv` for example?

Reply:
Because we implement right shift with udiv, but in that case we divide by two so this will never fail and the location is not needed here.

Reply:
I think I'd rather include the location in right shift as well then rather than making it optional here. Otherwise we have to make sure to uphold this invariant of only having it as None for this case

Reply:
Pinging on this conversation  -- I think its fine to make it non optional so that we do not need to explain the reasoning as to why UDiv is optional and SDiv is not

Reply:
@guipublic could we resolve this as your PR is really significant and important

Reply:
Done